File: app/src/main/java/club/ranleng/psnine/data/interceptor/RequestInterceptor.java
package club.ranleng.psnine.data.interceptor;

import java.io.IOException;

import okhttp3.Request;
import okhttp3.Response;

public class RequestInterceptor implements okhttp3.Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Request newRequest;
        newRequest = request.newBuilder()
                .addHeader("User-Agent", "Unofficial PSNine Android Application by RanKKI")
                .build();
        return chain.proceed(newRequest);
    }
}

File: app/src/main/java/xyz/rankki/psnine/data/http/interceptor/RequestInterceptor.kt
package xyz.rankki.psnine.data.http.interceptor

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import xyz.rankki.psnine.utils.Log

class RequestInterceptor : okhttp3.Interceptor {

    override fun intercept(chain: Interceptor.Chain?): Response {
        val newRequest: Request = chain!!.request()
                .newBuilder()
                .addHeader("User-Agent", "Unofficial PSNine Android Application by RanKKI")
                .build()
        Log.i("request ${newRequest.url()}")
        return chain.proceed(newRequest)
    }

}